lib/src/graph/package_graph.dart

```diff
@@ -168,9 +168,10 @@ class PackageGraph {
       return new Future.error(error, _lastUnexpectedErrorTrace);
     }
 
-    // If the last build completed with an error, complete the future with it.
-    if (!_lastResult.succeeded) {
-      return new Future.error(BarbackException.aggregate(_lastResult.errors));
+    // If any build completed with an error, complete the future with it.
+    var result = new BuildResult.aggregate(_results);
```

Comment:
AFAICT there is no 'next run' after going through this code path.

The `build` command is single pass, it only calls this once.